### PR TITLE
[SPARK-14504][SQL] Enable Oracle docker tests

### PR DESCRIPTION
**Pull request by lresende**
## What changes were proposed in this pull request?

Enable Oracle docker tests
## How was this patch tested?

Existing tests

### CODE REVIEW

**Comment by lresende** [comment]
This PR is dependent on PR #9893

**Comment by JoshRosen** [comment]
I just merged #9893 so this should be ready for a rebase.

**Comment by JoshRosen** [comment]
Ping RE: rebase.

**Comment by lresende** [comment]
Jenkins, retest this

**Comment by lresende** [comment]
Jenkins, retest this please.

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -364,12 +364,15 @@ object Flume {
 }
 
 object DockerIntegrationTests {
+  // Ignore checksum
+  checksums in update := Nil
```
Comment (JoshRosen) [inline project/SparkBuild.scala]: Why this change?

Comment (lresende) [inline project/SparkBuild.scala]: The jar was getting a warning for checksum on maven build, and failing on sbt build... let me try removing it now that I explicitly pointed where the jar is for sbt, wich might make this change obsolete.

**Comment by JoshRosen** [comment]
I'm a bit skeptical of some of the build hacks needed for this but out of pragmatism I'm going to merge this now and will just remove these tests later if this turns out to be unreliable.

**Comment by rxin** [comment]
These tests are too flaky. I've already seen a few failures.

We should disable them from the normal tests and maybe occasionally running them (via some trigger or just run it once before the release).

**Comment by rxin** [comment]
I just failed to build Spark locally once due to the docker oracle test.

**Comment by lresende** [comment]
@rxin, just trying to understand, is the oracle test the only one failing ? Or you are suggesting we move the whole docker based tests to a separate profile that would only run on Jenkins ?

**Comment by rxin** [comment]
I don't even think they should run on pull requests. Tests that require extensive external setup (or downloading things) in general are flaky.

**Comment by lresende** [comment]
@rxin Let me move them to a specific docker profile. But I would still run them on Jenkins, as the infrastructure is already setup there.

**Comment by rxin** [comment]
They have been flaky and causing other pull requests to fail. That's why we shouldn't run them on Jenkins.

**Comment by rxin** [comment]
We can run them via some trigger phrase though.

**Comment by lresende** [comment]
Ok, I will work with @JoshRosen on the trigger part.
